client/driver/docker.go

```diff
@@ -226,18 +236,22 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task) (do
 	}
 	hostConfig.NetworkMode = mode
 
-	// Setup port mapping (equivalent to -p on docker CLI). Ports must already be
-	// exposed in the container.
+	// Setup port mapping and exposed ports
 	if len(task.Resources.Networks) == 0 {
-		d.logger.Print("[WARN] driver.docker: No networks are available for port mapping")
+		d.logger.Print("[WARN] driver.docker: No network resources are available for port mapping")
 	} else {
+		// TODO add support for more than one network
 		network := task.Resources.Networks[0]
-		dockerPorts := map[docker.Port][]docker.PortBinding{}
+		publishedPorts := map[docker.Port][]docker.PortBinding{}
+		exposedPorts := map[docker.Port]struct{}{}
 
 		for _, port := range network.ListStaticPorts() {
-			dockerPorts[docker.Port(strconv.Itoa(port)+"/tcp")] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port)}}
-			dockerPorts[docker.Port(strconv.Itoa(port)+"/udp")] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port)}}
+			publishedPorts[docker.Port(strconv.Itoa(port)+"/tcp")] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port)}}
```

Comment:
@cbednarski So we are publishing both udp and tcp protocols for the same port? Now that we have the port block we can ask user to specify the protocol too,